keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,532 @@
+  Each container will have a listening port open to receive messages and also a process periodically sending messages to an remote address.
+  The received messages will be printed on stdout.
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create Pods in the new namespace
+
+
+  Test setup
+
+  - Create two namespaces, ns1 and ns2
+
+
+  Test steps
+
+  - Create two Pods in ns1
+  - Check the output to make sure they can communicate with each other
+  - Create two Pods, one in ns1, and the other in ns2
+  - Check the output to make sure they can't talk to each other.
+
+
+  Test teardown
+
+  - Delete the namespaces
+
+### Non-behavioral Criteria
+
+#### API server configuration
+
+  *Conformance Profile: Essential*
+
+  The following are required:
+
+  - All communication over mTLS/TLS
+    - apiserver talks to etcd via HTTPS
+    - controller-manager, scheduler talks to apiserver via HTTPS
+
+
+  The following should be disabled:
+
+  - insecure listening port
+  - basic-auth
+  - token-auth-file
+  - AlwaysAllow authorizer
+  - Admission control:
+    - AlwaysAllow
+    - NamespaceAutoProvision
+  - profiling
+
+  The following are required:
+
+  - RBAC authorizer
+  - Admission controls:
+    - PodSecurityPolicy
+    - NamespaceLifecycle
+
+##### Test methodology
+
+  The same test methodology is used as [Config File Permissions](#config-file-permissions).
+  For self-hosted Kubernetes cluster, the verification becomes easier,
+  as the control planes are managed as Kubernetes API objects which can be directly inspected by test code via Kubernetes API.
+
+#### Controller Manager
+
+  *Conformance Profile: Essential*
+
+  The following are required:
+
+  - use service account credentials
+
+##### Test methodology
+
+  Same as [API server configuration](#api-server-configuration)
+
+#### API server configuration - Restricted
+
+  *Conformance Profile: Restricted, Isolated*
+
+  The following admission controls are required:
+  - PodSecurityPolicy
+  - NamespaceLifecycle
+  - NamespaceExists
+  - DenyEscalatingExec
+  - DenyExecOnPrivileged
+  - EventRateLimit
+  - LimitRanger
+  - NodeRestriction
+  - PodTolerationRestriction
+  - AlwaysPullImages
+
+#### Use Sandboxed Isolation
+
+  *Conformance Profile: Sandboxed*
+
+  The container runtime must be something using HyperVisor or gVisor.
+
+##### Test methodology
+
+  Same as [API server configuration](#api-server-configuration), the command line
+  of `kubelet` should be verified to use the correct container runtime.
+  The list of supported container runtimes is TO BE DETERMINED.
+
+## Graduation Criteria
+
+- Conformance Profiles are well defined and finalized
+- Definition of security criteria are curated and documented
+- Tests are added to conformance test suite to cover security criteria and categorized according to conformance profiles
+
+## Alternatives
```

Comment:
Can you discuss how this compares with the CIS benchmark, and linters like [kube-bench](https://github.com/aquasecurity/kube-bench)

Reply:
Added in Alternatives.